src/components/input/inputContainer/inputContainer.api.ts

```diff
@@ -0,0 +1,56 @@
+import type { InputHTMLAttributes, ReactNode } from 'react';
+
+export type InputVariant = 'default' | 'warning' | 'critical';
+
+export interface IInputContainerProps {
+    /**
+     * Label of the input.
+     */
+    label?: string;
+    /**
+     * Variant of the input.
+     * @default default
+     */
+    variant?: InputVariant;
+    /**
+     * Help text displayed above the input.
+     */
+    helpText?: string;
+    /**
+     * Displays the optional tag when set to true.
+     */
+    isOptional?: boolean;
+    /**
+     * Displays the input as disabled when set to true.
+     */
+    isDisabled?: boolean;
+    /**
+     * Additional information displayed below the input.
+     */
+    infoText?: string;
+    /**
+     * Alert message displayed below the input.
+     */
+    alertMessage?: string;
+    /**
+     * Id of the input field.
+     */
+    id: string;
+    /**
+     * Children of the component.
+     */
+    children?: ReactNode;
```

Comment:
I see, yes we could use `PropsWithChildren` instead of `React.FC<Props>` on the component. On the other hand I'd prefer to specify/document all the props in one place and have this consistent on the repository. What do you think?

Reply:
I think extending the `IInputContainerProps` with `PropsWithChildren` should get the job done. That said, not exactly saving you much in terms of keystrokes anyway; it was just an observation